packages/tpetra/core/example/Finite-Element-Assembly/README.md

```diff
@@ -4,30 +4,56 @@ Finite Element Mesh Assembly
 
 Files
 ------
-- `typedefs.hpp`
+
+### Common Files
+- `fem_assembly_typedefs.hpp`
   - Type definitions for the FEM Assembly examples.
-- `Element.hpp`
+- `fem_assembly_Element.hpp`
   - Provides functions that provide a rough approximation of a 2D finite-difference stencil.
-- `MeshDatabase.hpp`
+- `fem_assembly_MeshDatabase.hpp`
   - Provides the core 2D mesh database for this set of examples.
-- `fem_assembly_InsertGlobalIndices_DP.cpp`
+- `fem_assembly_commandLineOpts.hpp`
+  - Provides helpers for processing command line options.
+- `fem_assembly_main`
+  - Contains the main() function for the application.
+
+### Finite Element Mesh Assembly Examples
+- `fem_assembly_InsertGlobalIndices_DP.hpp`
   - Constructs the crsGraph by looping over the _owned_ elements and inserting the representations of the
     connectivity of each element into the graph using their global ids and lets Tpetra handle the exchanges.
   - Performance:  :star:
   - Ease of use:  :star: :star: :star:
   - DynamicProfile
-- `fem_assembly_LocalElementLoop_DP.cpp`
+- `fem_assembly_LocalElementLoop_DP.hpp`
   - This method distinguishes owned from overlapping nodes.
     - Owned nodes are those that _only_ touch elements owned by the same process.
     - Overlapping nodes are nodes that touch elements owned by a different process.
   - Performance:  :star: :star:
   - Ease of use:  :star:
   - DynamicProfile
-- `fem_assembly_TotalElementLoop_DP.cpp`
+- `fem_assembly_TotalElementLoop_DP.hpp`
   - Each process contains information from both its owned elements and its ghost elements so mesh construction
     is possible without requiring communication.
   - Performance:  :star: :star: :star:
```

Comment:
We can keep them or toss them.  That part needs work anyhow since it was just a guesstimation on my part since the various assembly methods aren't something I've dealt with in the past.  The readme isn't supposed to be comprehensive -- The information in the guide that Tim's putting together will have much more detail.  I figured this should be a high-level description and then eventually provide links to the full docs once those get published somewhere.

Reply:
@william76 I'm OK with that :)